src/rbus/rbus_interval_sub.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * If not stated otherwise in this file or this component's Licenses.txt file
+ * the following copyright and licenses apply:
+ *
+ * Copyright 2016 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#define _GNU_SOURCE 1 //needed for pthread_mutexattr_settype
+#include "rbus_interval_sub.h"
+#include "rbus_config.h"
+#include "rbus_handle.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+#include <rtVector.h>
+#include <rtTime.h>
+#include <rtMemory.h>
+#include <rbuscore.h>
+
+#define ERROR_CHECK(CMD) \
+{ \
+  int err; \
+  if((err=CMD) != 0) \
+  { \
+    RBUSLOG_ERROR("Error %d:%s running command " #CMD, err, strerror(err)); \
+  } \
+}
+#define VERIFY_NULL(T)      if(NULL == T){ return; }
+
+rtVector gRecord = NULL;
+static pthread_mutex_t gMutex = PTHREAD_MUTEX_INITIALIZER;
+typedef struct sRecord
+{
+    int running;
+    pthread_mutex_t  mutex;
+    pthread_t        thread;
+    pthread_cond_t   cond;
+    rbusHandle_t     handle;
+    elementNode const* node;
+    rbusProperty_t   property;
+    rbusSubscription_t* sub;
+} sRecord;
+
+void rbusEventData_appendToMessage(rbusEvent_t* event, rbusFilter_t filter, int32_t interval, int32_t duration, int32_t componentId, rbusMessage msg);
+
+static void init_thread(sRecord* sub_rec)
+{
+    pthread_mutexattr_t attrib;
+    pthread_condattr_t cattrib;
+
+    sub_rec->running = 0;
+
+    ERROR_CHECK(pthread_mutexattr_init(&attrib));
+    ERROR_CHECK(pthread_mutexattr_settype(&attrib, PTHREAD_MUTEX_ERRORCHECK));
+    ERROR_CHECK(pthread_mutex_init(&sub_rec->mutex, &attrib));
+
+    ERROR_CHECK(pthread_condattr_init(&cattrib));
+    ERROR_CHECK(pthread_condattr_setclock(&cattrib, CLOCK_MONOTONIC));
+    ERROR_CHECK(pthread_cond_init(&sub_rec->cond, &cattrib));
+    ERROR_CHECK(pthread_condattr_destroy(&cattrib));
+}
+
+static void sub_Free(void* p)
+{
+    sRecord* rec = (sRecord*)p;
+    if(rec)
+    {
+        rbusProperty_Release(rec->property);
```

Comment:
can  u destroy rec->mutex and rec->cond also?

Reply:
Changes are done